camel/datahubs/clients/huggingface.py

```diff
@@ -0,0 +1,260 @@
+
+        try:
+            self.api.upload_file(
+                path_or_fileobj=temp_file,
+                path_in_repo=filepath,
+                repo_id=dataset_name,
+                repo_type="dataset",
+            )
+        finally:
+            if os.path.exists(temp_file):
+                os.remove(temp_file)
+
+    def update_records(
+        self,
+        dataset_name: str,
+        records: List[Record],
+        filepath: Optional[str] = "records/records.json",
+    ) -> None:
+        temp_file = f"{dataset_name.replace('/', '_')}_records.json"
+        existing_records = []
+
+        try:
+            downloaded_file_path = hf_hub_download(
+                repo_id=dataset_name,
+                filename=filepath,
+                repo_type="dataset",
+                token=self.token,
+            )
+            with open(downloaded_file_path, "r") as f:
+                existing_records = json.load(f)
+        except Exception as e:
+            logger.error(e)
+            raise ValueError(
+                f"Dataset '{dataset_name}' does not have an existing file to "
+                f"update. Use `add_records` first."
+            )
+
+        new_records = [record.model_dump() for record in records]
+        combined_records = {
+            rec["id"]: rec for rec in (existing_records + new_records)
+        }.values()
+
+        with open(temp_file, "w") as f:
+            json.dump(list(combined_records), f)
+
+        try:
+            self.api.upload_file(
+                path_or_fileobj=temp_file,
+                path_in_repo=filepath,
+                repo_id=dataset_name,
+                repo_type="dataset",
+            )
+        finally:
+            if os.path.exists(temp_file):
+                os.remove(temp_file)
+
+    def delete_record(
+        self,
+        dataset_name: str,
+        record_id: str,
+        filepath: Optional[str] = "records/records.json",
+    ) -> None:
+        r"""Deletes a record from the dataset.
+
+        Args:
+            dataset_name (str): The name of the dataset.
+            record_id (str): The ID of the record to delete.
+            filepath: The path to the file containing the records.
+        """
+        temp_file = f"{dataset_name.replace('/', '_')}_records.json"
+        existing_records = []
+
+        try:
+            downloaded_file_path = hf_hub_download(
+                repo_id=dataset_name,
+                filename=filepath,
+                repo_type="dataset",
+                token=self.token,
+            )
+            with open(downloaded_file_path, "r") as f:
+                existing_records = json.load(f)
+        except Exception:
+            raise ValueError(
+                f"Dataset '{dataset_name}' does not have an existing file to "
+                f"delete records from."
+            )
+
+        filtered_records = [
+            record for record in existing_records if record["id"] != record_id
+        ]
```

Comment:
there’s some duplicated logic for handling file downloads and JSON operations, how about refactoring common logic into a helper function